schema/v1.1.0/dataset_config.yaml

```diff
@@ -0,0 +1,405 @@
+    description: A glob class for finding files in the output / destination directory.
+    is_a: GeneralGlob
+
+  SourceGlob:
+    description: A glob class for finding files in the source directory.
+    is_a: GeneralGlob
+
+  SourceMultiGlob:
+    description: A glob class for finding files in the source directory (with multiple globs).
+    attributes:
+      list_globs:
+        description: The globs for the file.
+        multivalued: true
+        range: string
+        required: true
+
+  # TODO: LinkML doesn't support all the validation for sources, so we'll need to extend it more in Python / JSON Schema
+  DefaultSource:
+    description: A generalized source class with glob finders.
+    attributes:
+      destination_glob:
+        description: The glob object for the destination file.
+        range: DestinationGlob
+      source_glob:
+        description: The glob object for the source file.
+        range: SourceGlob
+      source_multi_glob:
+        description: The glob object for the source file (with multiple globs).
+        range: SourceMultiGlob
+      # Would also have literal here, but literal schema varies by class, so we have DefaultLiteralEntity instead
+      # Would also have parent_filters here, but sometimes parent_filters doesn't apply (e.g., Dataset),
+      # so we have SourceParentFiltersEntity instead
+
+  SourceParentFiltersEntity:
+    description: Used as a mixin with root-level classes that contain sources that can have parent filters.
+    attributes:
+      parent_filters:
+        description: Filters for the parent of a source.
+        range: SourceParentFilters
+
+  SourceParentFilters:
+    description: Filters for the parent of a source.
+    attributes:
+      include:
+        description: Include files for the parent of a source (regexes).
+        range: SourceParent
+      exclude:
+        description: Exclude files for the parent of a source (regexes).
+        range: SourceParent
+
+  SourceParent:
+    description: A filter for a parent class of a source. For a given attribute, it can only be used if the current class is a subclass of the attribute.
+    attributes:
+      # Note that some classes are commented out since they have no children as per IMPORTER_DEP_TREE
+      # dataset_keyphoto:
+      # frame:
+      # gain:
+      # key_image:
+      # raw_tilt:
+      # tiltseries:
+      annotation:
+        description: Include or exclude annotations.
+        multivalued: true
+        range: string
+      dataset:
+        description: Include or exclude datasets.
+        multivalued: true
+        range: string
+      run:
+        description: Include or exclude runs.
+        multivalued: true
+        range: string
+      tomogram:
+        description: Include or exclude tomograms.
+        multivalued: true
+        range: string
+      voxel_spacing:
+        description: Include or exclude voxel spacings.
+        multivalued: true
+        range: string
```

Comment:
Ok, technically not all parents are valid for all classes -- for example it makes no sense to filter a Run by Tomogram (because tomograms are children of runs). I think this is 100% fine for now, but if you have time to think about whether we should make the schema stricter, it would be more accurate.

Reply:
Sounds good, I'll definitely follow up with you about this soon!